src/app/wallets/update-on-chain-payments.ts

```diff
@@ -0,0 +1,113 @@
+import { BTC_NETWORK, ONCHAIN_SCAN_DEPTH } from "@config"
+
+import { InvalidWalletId } from "@domain/errors"
+import { paymentAmountFromNumber } from "@domain/shared"
+import { translateLedgerTxnToWalletTxn } from "@domain/wallets/tx-history"
+import { TxDecoder, UnconfirmedOnChainTxError } from "@domain/bitcoin/onchain"
+import { CouldNotFindTransactionError, LedgerTransactionType } from "@domain/ledger"
+
+import { OnChainService } from "@services/lnd/onchain-service"
+import { NotificationsService } from "@services/notifications"
+import { getNonEndUserWalletIds, LedgerService } from "@services/ledger"
+import {
+  AccountsRepository,
+  UsersRepository,
+  WalletsRepository,
+} from "@services/mongoose"
+import { DisplayCurrency } from "@domain/fiat"
+
+export const updateOnChainPaymentsByTxHash = async ({
+  txHash,
+  scanDepth = ONCHAIN_SCAN_DEPTH,
+  logger,
+}: {
+  txHash: OnChainTxHash
+  scanDepth?: ScanDepth
+  logger: Logger
+}): Promise<true | ApplicationError> => {
+  const onChainService = OnChainService(TxDecoder(BTC_NETWORK))
+  if (onChainService instanceof Error) return onChainService
+
+  const tx = await onChainService.lookupOnChainPayment({ txHash, scanDepth })
+  if (tx instanceof Error) return tx
+  if (tx.confirmations === 0) return new UnconfirmedOnChainTxError()
```

Comment:
Should we leave this comment in here somewhere still, or signal explicitly in the code somewhere that we don't know the txid? Is it still a problem that trigger can happen before we know the txid?
```
      // we have to return here because we will not know whose user the the txid belong to
      // this is because of limitation for lnd onchain wallet. we only know the txid after the
      // transaction has been sent. and this events is trigger before

Reply:
Added https://github.com/GaloyMoney/galoy/pull/1916/commits/b7e61a2c458d234872e57443603ea7bbedef45d8